src/main/java/tutorease/address/logic/commands/DeleteContactCommand.java

```diff
@@ -52,12 +54,23 @@ public CommandResult execute(Model model) throws CommandException {
         if (targetIndex.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
-
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+
+        deleteLessonsForStudent(model, personToDelete);
         model.deletePerson(personToDelete);
         return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, format(personToDelete)));
     }
 
+    private void deleteLessonsForStudent(Model model, Person student) {
```

Comment:
It is also possible to have this method in the LessonSchedule.java such that the method takes in a person and will delete it within itself and we do not need to extract the list from the class. Therefore we just have to call deleteLessonsOfPerson(person).